services/policies/pkg/service/event/service.go

```diff
@@ -0,0 +1,78 @@
+package eventSVC
+
+import (
+	"context"
+	"github.com/cs3org/reva/v2/pkg/events"
+	"github.com/owncloud/ocis/v2/ocis-pkg/log"
+	"github.com/owncloud/ocis/v2/services/policies/pkg/engine"
+)
+
+// Service defines the service handlers.
+type Service struct {
+	query  string
+	log    log.Logger
+	stream events.Stream
+	engine engine.Engine
+}
+
+// New returns a service implementation for Service.
+func New(stream events.Stream, logger log.Logger, engine engine.Engine, query string) (Service, error) {
+	svc := Service{
+		log:    logger,
+		query:  query,
+		engine: engine,
+		stream: stream,
+	}
+
+	return svc, nil
+}
+
+// Run to fulfil Runner interface
+func (s Service) Run() error {
+	ch, err := events.Consume(s.stream, "policies", events.StartPostprocessingStep{})
+	if err != nil {
+		return err
+	}
+
+	for e := range ch {
+		switch ev := e.Event.(type) {
+		case events.StartPostprocessingStep:
+			if ev.StepToStart != "policies" {
+				continue
+			}
+
+			env := engine.Environment{
+				Stage: engine.StagePP,
+				User:  *ev.ExecutingUser,
+				Resource: engine.Resource{
+					ID:   *ev.ResourceID,
```

Comment:
Question: Is it better to check for `nil` before dereferencing `ev.ExecutingUser` and `ev.ResourceId`? It will panic if these values are `nil`.

Reply:
i thought those values are guaranteed, i added the nil check!